File: server/src/module/product/product.js
import mongoose from 'mongoose';

const { Schema } = mongoose;

const productSchema = new Schema(
    {
        handle: {
            type: String,
            required: true,
        },
        title: {
            type: String,
            required: true,
        },
        body: String,
        vendor: {
            type: String,
            required: true,
        },
        category: {
            type: String,
            required: true,
        },
        tags: [String],
        price: {
            type: String,
            required: true,
        },
        photoUrl: {
            type: String,
            required: true,
        },
        videoUrl: {
            type: String,
            required: true,
        },
    },
    { timestamps: true }
);

const Product = mongoose.model('Product', productSchema);

export default Product;

File: jalwa/app/src/main/java/com/example/jalwa/data/model/Product.kt
package com.example.jalwa.data.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Product (
        @SerializedName("_id")
        val id: String,

        val handle: String,
        val title: String,
        val body: String? = null,
        val vendor: String,
        val type: String,
        val tags: List<String>,
        val price: String,

        @SerializedName("photoUrl")
        val photoURL: String,
        @SerializedName("videoUrl")
        val videoURL: String
) : Serializable

